pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStoreTableView.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.metadata.api;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+
+/**
+ * Defines metadata store tableview.
+ * MetadataStoreTableView initially fills existing items to its local tableview and eventually
+ * synchronize remote updates to its local tableview from the remote metadata store.
+ * This abstraction can help replicate metadata in memory from metadata store.
+ */
+public interface MetadataStoreTableView<T> {
```

Comment:
Regarding this interface, I found the following methods are only used in tests.

These methods are also never used except in tests.

```java
    T get(String key);
    int size();
    boolean isEmpty();
    Set<String> keySet();
    Collection<T> values();
    void forEach(BiConsumer<String, T> action);
```

Please remove them. Since this interface is mainly designed to be used internally by `ServiceUnitStateMetadataStoreTableViewImpl`, we'd better keep the interface as simple as possible so that we won't need to write many unnecessary tests because they're never used.

If you're going to verify some internal states in tests, you can just access the Impl's methods or fields directly.

The `org.apache.pulsar.broker.transaction.buffer.impl.TableView` written by me is a good example that it only retains necessary methods that are used internally.

Reply:
`T get(String key); ` is still required here. I will remove the others.